code/discovery.py

```diff
@@ -162,29 +288,62 @@ def readRuntimeFiles(path):
     return None
 
 
+def cudaCoresInformation(nvDevices, gpus):
+    
+    devices, libs = buildCudaCoreDockerCLI(nvDevices)
+    output = cudaCores(image, devices, libs, gpus)
+    if output != '':
+        information = cudaInformation(output)
+        return information
+    else:
+        return []
+
+
 def flow(runtime, hostFilesPath):
     runtime = searchRuntime(runtime, hostFilesPath)
 
+    runtimeFiles = {
+        'available': True,
+        'name': 'Graphics Processing Unit',
+        'vendor': 'Nvidia',
+        'classes': ['gpu'],
+        'identifier': identifier,
+        'additional-assets': runtime
+    }
+
     if runtime is not None:
+
         # GPU is present and able to be used
 
         if dockerVersion():
             logging.info('--gpus is available in Docker...')
 
         else:
             logging.info('--gpus is not available in Docker, but GPU usage is available')
+        
+        runtime['additional-assets'] = {'device-information': cudaCoresInformation(runtime['devices'], True)}
```

Comment:
did you try this against the testbed? does the `device-information` attr in Nuvla?

Reply:
Updated to reflect changes to Nuvla